### ISSUE TITLE
Do you support `init.lua`?

### ISSUE THREAD

**Issue by nyngwang** [vscode-neovim/vscode-neovim#768]
As title.

**Comment by davelosert** [comment]
Not an official answer, but:
I have my nvim configured with lua-scripts and an `init.lua` and it is working flawelessly.

Little tip: 
User `if(not vim.g.vscode)` to configure stuff that is only supposed to happen when running nvim natively.

**Comment by ninjarogue** [comment]
I tried the conditional `if (not vim.g.vscode) then ...` and everything kinda works except that the plugins are still used. I don't want my plugins to be integrated with vscode, any solutions?

**Comment by davelosert** [comment]
Hmmm does not happen for me.
I am using [packer](https://github.com/wbthomason/packer.nvim) and i also wrapped the init of those plugins I do not want in the condition above.

E.g. I installed the base64-plugin and did the following:

```lua
if(not vim.g.vscode)
  then
    local base16 = require "base16"
    base16(base16.themes["solarized-dark"], true)
end
```

Works quite well - but I am not a lua config or packer pro. I mainly use nvim in VSCode right now so it might not work for every plugin like this.

**Comment by lbowenwest** [comment]
packer has the `cond` option that you can pass to optionally load a plugin. I found the `disable` didn't work for dynamic variables. 

For instance I have 
```lua
use {
  'vimwiki/vimwiki',
  setup = [[require('config.vimwiki')]],
  cond = function() return not vim.g.vscode end
}
```
to only load vimwiki if not in vs code

**Comment by nyngwang** [comment]
@lbowenwest Thanks for your sharing! I will definitely try it later!

**Comment by amrox** [comment]
something I ran into that might be helpful to others -

It does not seem like setting Neovim Init Vim Paths (`vscode-neovim.neovimInitVimPaths.darwin`) explicitly to an `init.lua` worked.

However, if I just left the setting blank, it picked up my `~/.config/nvim/init.lua`.

**Comment by chrisands** [comment]
Sadly it doesn't work with `packer.nvim` because it compiles config. I couldn't make it work. Now I thinking why `vscode-neovim.neovimInitVimPaths.darwin` doesn't work as it suppose to.

**Comment by chrisands** [comment]
I found out that any plugin in `site/pack/*/start/*` will be autoloaded. I thought that it is impossible, but `packer` allows make plugins optional which places them into `site/pack/packer/opt`. But you can't just write `require('packer').loader('plugin_one', 'plugin_two')` after `startup` function. So I thought to use autocmd for it. And it actually worked pretty well:
```lua
if (vim.g.vscode) then
  vim.cmd([[
    augroup packer_load_lazy_vscode_plugins
      autocmd VimEnter * PackerLoad vim-sandwich
      autocmd VimEnter * PackerLoad vim-easymotion
      autocmd VimEnter * nmap <C-j> <Plug>(easymotion-j)
      autocmd VimEnter * nmap <C-k> <Plug>(easymotion-k)
    augroup end
  ]])
else
  vim.cmd([[
    augroup packer_load_lazy_plugins
      autocmd!
      autocmd VimEnter * PackerLoad vim-sandwich
      autocmd VimEnter * PackerLoad autocmd-lua
    augroup end
  ]])
end

require('packer').startup(function(use)
  use {
    -- Package manager
    'wbthomason/packer.nvim',
  }
  -- all other plugins should be optional

end

```

**Comment by nyngwang** [comment]
@chrisands 
TL;DR: Thanks for the info.
I asked this on Nov. 6, and never try it. Now I realize that it was a wise choice, since I has been using packer for so long.

**Comment by stefanwatt** [comment]
I can't get vscode-neovim to respect my init.lua. Doesn't matter if I link the file in vscode settings or leave it blank. Only loading keymaps, too.
`if (vim.g.vscode)then
  require "user.keymaps"
else 
...
end`
But keymaps don't work.
Can someone help ?

**Comment by diz517** [comment]
I tried amrox's solution, works for me.  Hope this helps
```
 if (vim.g.vscode) then
    -- VSCode extension
    vim.cmd([[source XXXXX.vim]])
else
    -- ordinary neovim
  ...
end
```

**Comment by chrisands** [comment]
FYI currently there is issue which breaks optionals loading https://github.com/wbthomason/packer.nvim/issues/865
But you can use fork until it is merged
```lua
require('packer').startup(function(use)
  use({
    -- Package manager
    "ifplusor/packer.nvim", -- using fork with branch that fixes loading issue
    opt = false,
    branch = "fix-sequencing",
  })

  use({
    -- Fancier statusline
    "nvim-lualine/lualine.nvim",
    opt = true, -- Plugin required to be optional to use condition
    cond = function()
      return not vim.g.vscode
    end,
    after = { "onedark.nvim" },
    requires = { "kyazdani42/nvim-web-devicons", opt = true },
    config = function()
      require("lualine").setup()
    end,
  })
end
```

**Comment by JoostScheffer** [comment]
@lbowenwest Thank you so much for sharing your solution
